### PR TITLE
Add repo README and VSCode extension recommendations

### PR DESCRIPTION
**Pull request by wangannie**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

This PR introduces developer-experience improvements:
- Introduced the main README including all setup steps
  - A lot of this content was pulled from @alliu879's [Repo Setup Notion page](https://www.notion.so/calblueprint/Local-Setup-ca26e25d171b499f9764c778b24fe0b3), with supplements from the [Healthy Corners Getting Started guide](https://healthycorners-rewards.netlify.app/customer/getting-started.html)
- Added VSCode workspace extension recommendations, which include ESLint, Prettier, and GitLens. This should make it easier for new/future developers to set up their dev environments.

Check out the preview of the README here: https://github.com/calblueprint/chinese-newcomers/tree/wannie/readme-extensions#readme
## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

Run through all the README steps ([preview here](https://github.com/calblueprint/chinese-newcomers/tree/wannie/readme-extensions#readme)) and make sure everything works as expected.

I'd appreciate feedback on clarity/structure/wording in the README, as well as any suggestions for things that I should add!

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- As we come across additional setup bugs/common issues, add them to the README
- Continue adding important VSCode extensions to the workspace file as needed

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"
https://code.visualstudio.com/docs/editor/extension-marketplace#_workspace-recommended-extensions
https://dev.to/askrishnapravin/recommend-vs-code-extensions-to-your-future-teammates-4gkb
https://docs.npmjs.com/downloading-and-installing-node-js-and-npm

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
Basically a mirror of this PR for v4v: https://github.com/calblueprint/vouchers4veggies/pull/27

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."

CC: @surajr10

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+### Prerequisites
+
+Check your installation of `npm` and `node`:
+
+```sh
+node -v
+npm -v
+```
+
+This project requires node version 16.17.0.
```
Comment (wangannie) [inline README.md:18]: @surajr10 this is based on the fact that i couldn't get yarn install to work with 12, 14, or 18, and @anniehuo's worked on 16.17.0 If you know other versions also work, feel free to suggest and I'll update.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+   5. Install project dependencies. This command installs all packages from [`yarn.lock`](yarn.lock).
+         ```sh
+         yarn install
+         ```
+2. Set up secrets:
+   1. In the project's root directory (`chinese-newcomers/`), create a new file named `.env`
+        ```sh
+        touch .env
+        ```
+   2. Copy the credentials from [Blueprint's internal Notion](https://www.notion.so/calblueprint/Firebase-Environment-Variables-c30a053fba4c47559f9f4944e4962a9f) (access is required) and paste them into the `.env` file.
```
Comment (alliu879) [inline README.md]: replace this v4v link w cnsc link: https://www.notion.so/calblueprint/Environment-Variables-2d24bb3a8b8c4b369fa620cbe1c8fe49?pvs=4

**Review by alliu879 — APPROVED** [review]
lgtm! just one small change needed

File: README.md (hunk 1 above)
Comment (surajr10) [reply README.md:18]: yea 18 didn't work for me either but 16.17.0 did so we can stick with that

File: README.md (hunk 2 above)
Comment (surajr10) [reply README.md]: fixed

**Review by surajr10 — APPROVED** [review]
lgtm! tyty
